lib/fencing/st_client.c

```diff
@@ -2565,7 +2565,7 @@ stonith__sort_history(stonith_history_t *history)
         tmp = hp->next;
         if ((hp->state == st_done) || (hp->state == st_failed)) {
             /* sort into new */
-            if ((!new) || (hp->completed > new->completed)) {
+            if ((!new) || (hp->completed >= new->completed)) {
```

Comment:
Do we need to do this in the test below this as well?

Reply:
> Do we need to do this in the test below this as well?

I didn't understand the meaning of your point.
What do you point out?

Reply:
@HideoYamauchi He's referring to just a couple lines down where there's this:
```
                      if ((!np->next) || (hp->completed > np->next->completed)) {
```
And yeah it does seem like that test should probably be changed too.

Reply:
Hi clumens,
Thanks!

No, there is no need to change the conditions in this part.
Because the processing at the same time is always processed by the changed condition.

```
                       if ((!new) || (hp->completed >= new->completed)) {
```

Anything processed under the following conditions corresponds to a search for a sorted list, so there is no need to change the conditions.

```
                       if ((!np->next) || (hp->completed > np->next->completed)) {
```


Best Regards,
Hideo Yamauchi.

Reply:
@HideoYamauchi ,

Thanks for the clarification. Looking at the code again, I see that the original list to be sorted is a hash table iteration over stonith_remote_op_list (in stonith_local_history_diff_and_merge()). Since the hash table has no inherent order, how can we know what order two operations in the same second should be in? I don't see how we can know whether we should use > or >=.

Reply:
Hi Ken,

> @HideoYamauchi ,
> 
> Thanks for the clarification. Looking at the code again, I see that the original list to be sorted is a hash table iteration over stonith_remote_op_list (in stonith_local_history_diff_and_merge()). Since the hash table has no inherent order, how can we know what order two operations in the same second should be in? I don't see how we can know whether we should use > or >=.

Your opinion is correct.
In my simple test, I didn't get any Ghashtable order issues, but in reality, if stonith_fence_history_trim () etc. occur frequently, I think the order will be out of order.
To be precise, in stonith_remote_op_list using Ghashteble, if the accuracy of completed is the same, my PR will not be able to handle it.

Many thanks,
Hideo Yamauchi.